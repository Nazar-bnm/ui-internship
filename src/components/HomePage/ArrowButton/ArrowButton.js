import React from 'react';
import PropTypes from 'prop-types';
import './ArrowButton.scss';


const arrowButton = (props) => {
  const {
    type, onClick, className,
  } = props;
  return (
    <span onClick={onClick} className={className}>
      {type && <i className={`angle ${type} icon arrowButton`}></i>}
    </span>
  );
};

arrowButton.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func,
  className: PropTypes.string,
};

arrowButton.defaultProps = {
  left: false,
  right: false,
  clicked: () => {},
  className: '',
};

export default arrowButton;
