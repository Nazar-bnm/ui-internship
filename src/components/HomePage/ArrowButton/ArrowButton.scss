@import '../../../scss/global_styles.scss';

i.arrowButton {
  color: #fff;
  background-color: $emperor;
  height: 2rem;
  width: 2rem;
  font-size: 2rem;
  opacity: 0.2;
  margin: 0;
  cursor: pointer;
  border-radius: 50%;
}