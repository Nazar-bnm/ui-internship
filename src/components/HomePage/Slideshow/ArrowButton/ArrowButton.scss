@import '../../../../scss/global_styles';

i.arrow-button__icon {
  opacity: 0.4;
  margin: 0;
  width: 2rem;
  height: 2rem;
  border-radius: 50%;
  background-color: $emperor;
  cursor: pointer;
  font-size: 2rem;
  color: $alto;
  
  &:hover,
  &:active {
    opacity: 0.6;
  }
}