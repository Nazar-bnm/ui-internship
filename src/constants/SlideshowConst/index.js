export const ANIMATION_NAMES = {
  CAROUSEL: 'carousel',
  FADE: 'fade',
  ZOOM_IN: 'zoom-in',
  ZOOM_OUT: 'zoom-out'
};

export const ARROW_BUTTON_TYPES = {
  LEFT: 'left',
  RIGHT: 'right'
};
