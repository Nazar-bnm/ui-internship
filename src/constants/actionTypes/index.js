// Counter
export const INCREMENT = 'INCREMENT';
export const DECREMENT = 'DECREMENT';
// Wish list
export const ADD_TO_WISHLIST = 'ADD_TO_WISHLIST';
export const REMOVE_FROM_WISHLIST = 'REMOVE_FROM_WISHLIST';
export const CHANGE_QUANTITY = 'INCREMENT_QUANTITY';
export const REMOVE_ITEM_FROM_CART = 'REMOVE_ITEM_FROM_CART';
// Filters
export const CHECK_CHECKBOX = 'CHECK_CHECKBOX';
export const RESET_FILTERS = 'RESET_FILTERS';
export const SET_PRICE_RANGE = 'SET_PRICE_RANGE';
// Products pagination
export const FETCH_PRODUCTS_PENDING = 'FETCH_PRODUCTS_PENDING';
export const FETCH_PRODUCTS_SUCCESS = 'FETCH_PRODUCTS_SUCCESS';
export const FETCH_PRODUCTS_ERROR = 'FETCH_PRODUCTS_ERROR';
export const CHANGE_ITEMS_NUMBER_ON_PAGE = 'CHANGE_ITEMS_NUMBER_ON_PAGE';
export const CHANGE_SORTING_ORDER = 'CHANGE_SORTING_ORDER';
export const CHANGE_ORDER_TYPE = 'CHANGE_ORDER_TYPE';
export const GET_PRODUCTS_SUCCESS = 'GET_PRODUCTS_SUCCESS';
export const GET_PRODUCTS_ERROR = 'GET_PRODUCTS_ERROR';
export const NOTIFICATION_SHOW = 'NOTIFICATION_SHOW';
export const NOTIFICATION_HIDE = 'NOTIFICATION_HIDE';
export const ADD_TO_CART = 'ADD_TO_CART';
export const NOTIFICATION_ITEM_ADD_TO_LIST = 'NOTIFICATION_ITEM_ADD_TO_LIST';
export const NOTIFICATION_ITEM_REMOVE_FROM_LIST = 'NOTIFICATION_ITEM_REMOVE_FROM_LIST';

// Blog page
export const GET_BLOG_ITEMS_SUCCESS = 'GET_BLOG_ITEMS_SUCCESS';
export const GET_BLOG_ITEMS_ERROR = 'GET_BLOG_ITEMS_ERROR';

// Cart
export const SET_TOTAL_COUNT = 'SET_TOTAL_COUNT';
