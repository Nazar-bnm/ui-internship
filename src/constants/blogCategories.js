export const blogCategories = ['celebrity style', 'fashion shows', 'causal events', 'shopping'];
