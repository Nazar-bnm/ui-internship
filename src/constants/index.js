export const defaultImages = [
  { src: 'src/assets/img/content/new4.png' },
  { src: 'src/assets/img/content/new2.png' },
  { src: 'src/assets/img/content/new3.png' },
  { src: 'src/assets/img/content/new1.png' }
];
