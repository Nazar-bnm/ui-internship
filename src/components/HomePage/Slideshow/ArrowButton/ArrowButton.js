import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { ARROW_BUTTON_TYPES } from '../../../../constants/SlideshowConst';

import './ArrowButton.scss';

const CN = 'arrow-button';
const { LEFT, RIGHT } = ARROW_BUTTON_TYPES;

const arrowButton = ({ type, onClick, className }) => (
  <span onClick={onClick} className={className}>
    <i className={cx('angle', [type], 'icon', `${CN}__icon`)} />
  </span>
);

arrowButton.propTypes = {
  type: PropTypes.oneOf([LEFT, RIGHT]),
  onClick: PropTypes.func,
  className: PropTypes.string
};

arrowButton.defaultProps = {
  type: '',
  onClick: () => {},
  className: ''
};

export default arrowButton;
