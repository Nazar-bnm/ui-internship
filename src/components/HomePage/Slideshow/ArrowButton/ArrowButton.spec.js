import React from 'react';
import { shallow } from 'enzyme';

import ArrowButton from './ArrowButton';
import { ARROW_BUTTON_TYPES } from '../../../../constants/SlideshowConst';

const { LEFT } = ARROW_BUTTON_TYPES;

describe('<ArrowButton />', () => {
  it('should make a snapshot of ArrowButton component', () => {
    const wrapper = shallow(<ArrowButton onClick={() => {} } type={LEFT} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke click handler on click', () => {
    const mockedClickHandler = jest.fn(() => {});
    const wrapper = shallow(<ArrowButton onClick={mockedClickHandler} type={LEFT} />);

    wrapper.find('span').first().simulate('click');
    expect(mockedClickHandler.mock.calls.length).toBe(1);
  });
});
