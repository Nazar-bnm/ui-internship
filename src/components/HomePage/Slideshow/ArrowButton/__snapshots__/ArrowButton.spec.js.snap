// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowButton /> should make a snapshot of ArrowButton component 1`] = `
<span
  className=""
  onClick={[Function]}
>
  <i
    className="angle left icon arrow-button__icon"
  />
</span>
`;
